### ISSUE TITLE
a long command (with %f with many files selected) fails silently

### ISSUE THREAD

**Issue by aleksejrs** [vifm/vifm#1017]
In June, I finally updated vifm for the fix of #912, but I am not sure it worked. Now such commands just fail silently, even if I start vifm with bash.

**Comment by xaizek** [comment]
Thanks, not sure if I have actually tested running a long command with `fish` then, maybe I thought that `$SHELL` was updated when it wasn't.  The problem is:
```
[~]$ /usr/bin/bash -c "eval \"\$0\$1\"" echo ' bash works'
bash works
[~]$ /usr/bin/fish -c "eval \"\$0\$1\"" echo ' fish does not'
[~]$
```

It's like arguments for `eval` are expected to be a part of `-c` parameter, therefore making it impossible to work around argument length limit.

**Comment by aleksejrs** [comment]
man fish:

>       -c or --command=COMMAND
>              Evaluate the specified commands instead of reading from the commandline, passing  additional  positional  arguments through $argv.


`/usr/bin/fish -c "eval \"\$argv\"" echo ' fish seems to work'`

**Comment by xaizek** [comment]
> `/usr/bin/fish -c "eval \"\$argv\"" echo ' fish seems to work'`

I thought `$argv` is something implicit.

> Note that fish has different and configurable characters to escape though. E.g., in a terminal, '$' or '?' may or may not need escaping.

So even if I decide to create a special case for this backward thing, it still won't actually work?

**Comment by aleksejrs** [comment]
> So even if I create a special case for this backward thing, it still won't actually work?
No idea.

**Comment by xaizek** [comment]
> > So even if I create a special case for this backward thing, it still won't actually work?
>
> No idea.

By the way, I didn't mean that it won't work at all.  I does work inside Vifm, thanks for showing how to use it, just thinking about other potential pitfalls.

> When a parameter includes an unquoted

That makes it sound less problematic, because they will be quoted in parameters and that might be handled consistently regardless of the settings.  Escaping in the commands above are present only because of running them in a shell which isn't the case when invoking a command from Vifm.

I don't like special-casing to work around deliberate bad decisions, but it can be added.

**Comment by aleksejrs** [comment]
> because they will be quoted in parameters

Yeah, "?" is a normal character when the string is quoted. I just enabled qmark-noglob though, so it won't be a problem any more (really shouldn't have let it stay off because of a missing `.fish` extension).

**Comment by xaizek** [comment]
> I just enabled qmark-noglob though, so it won't be a problem any more (really shouldn't have let it stay off because of a missing `.fish` extension).

The non-POSIX behaviour is not expanding it, funny that `fish` "deprecated" it.  I just thought that escaping `?` might result in some "unsupported escape" error, but it doesn't.
